### PR TITLE
🗞️ Created `packages/oapp-alt-evm` and added `EndpointV2Alt` to TestHelper

### PR DESCRIPTION
**Pull request by shankars99**
OAppAlt is OApp but with alt-endpoint support. This contains a different OAppSender as the fee is paid in ERC20 instead of native token. 

TestHelper also did not have support for EndpointV2Alt. Added a function that spins up endpoints with erc20s as the payment token that does not affect the current implementations of TestHelper testing.

Added testing for OAppAlt.

### CODE REVIEW

File: .changeset/smart-plants-hide.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@layerzerolabs/test-devtools-evm-foundry": major
+"@layerzerolabs/oapp-alt-evm": major
```
Comment (janjakubnanista) [inline .changeset/smart-plants-hide.md:3]: This will immediatelly bump the version to `1.0.0` which is not what we want I believe

**Review by janjakubnanista — CHANGES_REQUESTED** [review]

File: .changeset/smart-plants-hide.md (hunk 1 above)
Comment (shankars99) [reply .changeset/smart-plants-hide.md]: just added a minor version for devtools

File: packages/oapp-alt-evm/turbo.json
Comment (janjakubnanista) [inline packages/oapp-alt-evm/turbo.json:1]: Please delete - this is a mistake that is being copied over and over, I am just creating a PR to solve this.

File: packages/oapp-alt-evm/package.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      "imports": "./artifacts/*.json"
+    }
+  },
+  "files": [
+    "artifacts/**/*",
+    "contracts/**/*",
+    "test/**/*"
+  ],
+  "scripts": {
+    "build": "$npm_execpath compile",
```
Comment (janjakubnanista) [inline packages/oapp-alt-evm/package.json:31]: Please delete the `build` command: 

- `build` command builds JS artifacts and depends on `compile` (see `turbo.json`)
- `compile` command does any contract/program compilation

File: packages/oapp-alt-evm/package.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    "clean": "rm -rf .turbo cache artifacts",
+    "compile": "$npm_execpath forge build",
+    "test": "$npm_execpath forge test"
```
Comment (janjakubnanista) [inline packages/oapp-alt-evm/package.json:34]: Please remove the `$npm_execpath`, `forge` is a standalone binary

**Review by janjakubnanista — CHANGES_REQUESTED** [review]

**Pull request closed without merging**